src/test/scala/firrtlTests/CInferMDirSpec.scala

```diff
@@ -70,9 +70,8 @@ circuit foo :
 """.stripMargin
 
     val annotationMap = AnnotationMap(Nil)
-    val writer = new java.io.StringWriter
-    compile(CircuitState(parse(input), ChirrtlForm, Some(annotationMap)), writer)
+    val res = compileAndEmit(CircuitState(parse(input), ChirrtlForm, Some(annotationMap)))
     // Check correctness of firrtl
-    parse(writer.toString)
+    parse(res.emittedCircuitOption.get.value)
```

Comment:
Can you add a helper method to do this? Because its done all over the place in testing, like `res.getEmittedCircuit`?